model/project.rb

```diff
@@ -121,6 +121,99 @@ def create_api_key(used_for: "inference_endpoint")
     ApiKey.create_with_id(owner_table: Project.table_name, owner_id: id, used_for: used_for)
   end
 
+  def generate_vm_options(account:)
```

Comment:
Is it possible to move this method to routes/helpers/vm.rb? The API authorization revamp is going to rely on authorization related decisions made in Clover.  Additionally, this is being added purely for UI concerns and not for model-level concerns.

Reply:
I put this to project model because the options we display would differ based on the project. I'm also planning to use it for form validations(currently it is also done manually for each option) and possibly in the future we can return the option tree as API response so that users can know which options are available to them.

The good thing is all these use cases also fall in Clover's responsibility boundaries, so I don't see any problem with moving it there.

Reply:
OK, moved them for VM and private subnet. I'll move others too as we merge them.